src/redis/fixed_cache_impl.go

```diff
@@ -66,47 +69,60 @@ func (this *fixedRateLimitCacheImpl) DoLimit(
 
 		// Use the perSecondConn if it is not nil and the cacheKey represents a per second Limit.
 		if this.perSecondClient != nil && cacheKey.PerSecond {
-			if perSecondPipeline == nil {
-				perSecondPipeline = Pipeline{}
+			value, err := this.perSecondClient.GetValue(cacheKey.Key)
+			if err != nil {
+				logger.Error(err)
 			}
-			pipelineAppend(this.perSecondClient, &perSecondPipeline, cacheKey.Key, hitsAddend, &results[i], expirationSeconds)
+			results[i] = value
 		} else {
-			if pipeline == nil {
-				pipeline = Pipeline{}
+			value, err := this.client.GetValue(cacheKey.Key)
+			if err != nil {
+				logger.Error(err)
 			}
-			pipelineAppend(this.client, &pipeline, cacheKey.Key, hitsAddend, &results[i], expirationSeconds)
+			results[i] = value
 		}
 	}
 
-	if pipeline != nil {
-		checkError(this.client.PipeDo(pipeline))
-	}
-	if perSecondPipeline != nil {
-		checkError(this.perSecondClient.PipeDo(perSecondPipeline))
-	}
-
 	// Now fetch the pipeline.
 	responseDescriptorStatuses := make([]*pb.RateLimitResponse_DescriptorStatus,
 		len(request.Descriptors))
 	for i, cacheKey := range cacheKeys {
 
-		limitAfterIncrease := results[i]
-		limitBeforeIncrease := limitAfterIncrease - hitsAddend
+		limitBeforeIncrease := uint32(results[i])
```

Comment:
yes this is intended. since we are not increasing in this function, so it's should be the result that we get from redis. We will increase the value in increaseAsync function below.